src/components/terminal.rs

```diff
@@ -2,13 +2,16 @@ use crate::types::terminal_content::TerminalContent;
 use leptos::{ev::SubmitEvent, html::Input, *};
 use std::time::Duration;
 
+// TODO - Tech Debt: Find a way to not allow the whole
+// component but rather the single closure
+#[allow(clippy::redundant_closure)]
 #[component]
 pub fn Terminal<F>(
     cx: Scope,
-    // TODO - Tech Debt: Find a way to pass terminal in as mut
     terminal: TerminalContent,
```

Comment:
Is this supposed to be reactive/change later on? It seems like `TerminalContent` is only for one level, so I would assume this is updated? I would consider passing a `Signal<TerminalContent>` if this is the case.